### PR TITLE
HPCC-16406 Document BLAS Support in Std Library

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>
@johnholt Please review for technical accuracy
@g-pan Please review for grammar, etc 
A PDF is available at 
http://10.176.32.6:8080/view/Docs/job/JimsRepoBuilder/57/CE-DOCS=Ubuntu-14.04-amd64-2/artifact/build-docs/Release/docs/ECLStandardLibraryReference-6.2.0-rc5.pdf#page=188

### CODE REVIEW

File: docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml [hunk 1]
```diff
@@ -0,0 +1,1224 @@
...
+      </tgroup>
+    </informaltable>
+
+    <para><emphasis role="bold">ICellFunc </emphasis>is the function prototype
+    for Apply2Cells.</para>
+
+    <para>Example:</para>
+
+    <programlisting format="linespecific">IMPORT STD;
+STD.BLAS.ICellFunc(42,2,3);
```
Comment (johnholt) [inline docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml]: This is not a good example.
Use REAL8 my_func(Std.BLAS.Types.value_t v, Std.BLAS.Types.dimension_t x, Std.BLAS.Types.dimension_t y) := 1/v;  //set element to the reciprocal value
as a realistic example.

File: docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml [hunk 2]
```diff
@@ -0,0 +1,1224 @@
...
+          <row>
+            <entry>Return:<emphasis></emphasis></entry>
+
+            <entry>The triangular matrix requested</entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </informaltable>
+
+    <para>The <emphasis role="bold">dgetf2</emphasis> function computes the
```
Comment (johnholt) [inline docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml]: change dgetf2 to dpotf2

File: docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml [hunk 3]
```diff
@@ -0,0 +1,1224 @@
...
+    Cholesky factorization of a real symmetric positive definite matrix A. The
+    factorization has the form A = U**T * U , if UPLO = 'U', or * A = L *
```
Comment (johnholt) [inline docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml]: Looks like the comment block was bad. Change UPLO=A to tri set to Triangle.Upper and UPLO = L to Triangle.Lower

File: docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml [hunk 4]
```diff
@@ -0,0 +1,1224 @@
...
+          <row>
+            <entry>Return:<emphasis></emphasis></entry>
+
+            <entry>The vector</entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </informaltable>
+
+    <para>The <emphasis role="bold">make_vecror</emphasis> function generates
```
Comment (johnholt) [inline docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml]: change vecror to vector

File: docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml [hunk 5]
```diff
@@ -0,0 +1,1224 @@
...
+          <row>
+            <entry>Return:<emphasis></emphasis></entry>
+
+            <entry>The trace (sum of the diagonal entries) </entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </informaltable>
+
+    <para>The <emphasis role="bold">make_vecror</emphasis> function computes
```
Comment (johnholt) [inline docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml]: Change make_vecror to trace

**Review by johnholt — COMMENTED** [review]
Some minor errors found.

File: docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml (hunk 3 above)
Comment (JamesDeFabia) [inline docs/ECLStandardLibraryReference/SLR-Mods/BLAS.xml]: @johnholt  I am not sure i understand this comment

**Comment by JamesDeFabia** [comment]
@johnholt Please look again. Edits made

**Comment by JamesDeFabia** [comment]
@johnholt Final edits made, please review once again
PDF Version is available at: http://10.176.32.6:8080/view/Docs/job/JimsRepoBuilder/CE-DOCS=Ubuntu-14.04-amd64-2/lastSuccessfulBuild/artifact/build-docs/Release/docs/ECLStandardLibraryReference-6.2.0-rc5.pdf#page=188

**Comment by johnholt** [comment]
We recently (like Friday) changed the name of the Extract_Tri function to extract_tri in the code.  It would probably look better in the documentation to be extract_tri.

On the first page of the BLAS document, would you please add as a second paragraph the description of column major versus row major?  I have a suggested text:
{quote}
The BLAS functions use the column major mapping for the storage of a matrix.  This is the mapping used in Fortran, and has the entries of the first column followed by the entries of the second column.  This is the transpose of the row major form commonly used in the C language where the entries of the first row are followed by the entries of the second row.
{quote}

**Comment by JamesDeFabia** [comment]
@johnholt I made the final edits you suggested

**Comment by g-pan** [comment]
I approve.

**Pull request merged**